tests/lib/test_Filter.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from pathlib import Path
+
+from onyo.lib import Filter, OnyoInvalidFilterError
+
+
+@pytest.mark.parametrize('filt', ['type=laptop', 'key=value', 'foo=<unset>'])
+def test_filter(filt: str) -> None:
+    class Repo:
```

Comment:
You seem to emulate the Repo class, which I think is awkward, and makes it impossible later to add `from onyo.lib import Repo`. Is there a reason why you do not use `@pytest.mark.repo_contents`?

This function is missing a doc string, which would make it way easier to have an idea what this test is trying to verify before reading through the different if/else, the little class, and such.

Reply:
That is not true. I overwrite the `Repo` class only from within the scope of that test. It is not overwritten elsewhere.

Unfortunately `Filter` (which needs to use `Repo._read_assets()`) requires access to the current repo instance, because all asset manipulation is done in there. This would not be necessary if asset manipulation were done via an `Asset` class (I've mentioned this case before in matrix), because then an `Asset` object would be passed to `Filter`, instead of a `Path` object. 

However, `Filter` _only_ uses `Repo._read_asset()` and in this test it does not even need that, so adding the overhead of generating and initializing a repo for these tests is unnecessary. Since it still requires a `Repo` on initialization (and using `None` as a possible type for tests would create further issues, including very relevant pyre complaints) I instead give it a mock `Repo` that doesn't require initialization.